src/hotspot/share/runtime/objectMonitor.hpp

```diff
@@ -235,7 +235,7 @@ class ObjectMonitor : public CHeapObj<mtObjectMonitor> {
   // to the ObjectMonitor reference manipulation code:
   //
   #define OM_OFFSET_NO_MONITOR_VALUE_TAG(f) \
-    ((in_bytes(ObjectMonitor::f ## _offset())) - markWord::monitor_value)
+    ((in_bytes(ObjectMonitor::f ## _offset())) - (unsigned)markWord::monitor_value)
```

Comment:
This makes it int vs uint (-Wsign-conversion warning), and assumes the range of monitor_value.
One option is to narrow the type of monitor_value.  Option 2, cast the lhs to uintptr_t.  Option 3, use offset_of instead of signed ByteSize.

Reply:
I looked at option 1, but monitor_value is used for byte operations, and it seemed unsafe to make it smaller than uintptr_t, eg:

oops/markWord.hpp:    return markWord(tmp | monitor_value);

The -Wconversion warning is complaining about OM_OFFSET_NO_MONITOR_VALUE_TAG is an long unsigned int because the expression is promoted to the size of markWord::monitor_value without the cast.

Option 3 turns the address into a 64 bit int, but the Address constructor expects a 32 bit displacement.

   88 |   __ movptr(Address(mon, OM_OFFSET_NO_MONITOR_VALUE_TAG(owner)), r15_thread);

Should this convert to int instead of unsigned instead?

Reply:
Yes int would be more correct.  The value could be negative if the order of the fields in ObjectMonitor were changed.